docs/teams/teams_concepts.md

```diff
@@ -0,0 +1,43 @@
+# Understanding Livebook Teams concepts
```

Comment:
Should we talk about secrets and file systems?

Reply:
Yes. But I think we can do that in another PR.

I tried to focus only on what was needed for the “deploy app” tutorial at this moment. So we can ship it sooner than later.